internal/service/entitlement.go

```diff
@@ -116,6 +116,26 @@ func (s *entitlementService) CreateEntitlement(ctx context.Context, req dto.Crea
 			Mark(ierr.ErrValidation)
 	}
 
+	// For metered features, check if it's a bucketed max meter
+	if feature.Type == types.FeatureTypeMetered {
+		meter, err := s.MeterRepo.GetMeter(ctx, feature.MeterID)
+		if err != nil {
+			return nil, err
+		}
+
+		// Entitlements are restricted for bucketed max meters
+		if meter.Aggregation.Type == types.AggregationMax && meter.Aggregation.BucketSize != "" {
```

Comment:
this condition can be wrapped as a function in meter domain - can be used in all if checks
```
func (m *Meter) IsBucketedMeter() {
    if m.Aggregation.Type == types.AggregationMax && m.Aggregation.BucketSize != ""
}
```
or maybe just check the bucketSize